### ISSUE TITLE
Profile Patching and properties (v1.1.X)

### ISSUE THREAD

**Issue by DavidAsb** [zac-schutt/KlaviyoSharp#21]
It looks like the Profile and ProfilePatch structures are different from the spec. ( https://developers.klaviyo.com/en/v2023-07-15/reference/update_profile )

Any attempt to update/append properties on an existing profile via MetaProperties returns 'invalid field type'
Changing MetaProperties to remove the list appears to fix this and the append now adds to the list of an existing property.

```
    //public List<PatchProperties> PatchProperties { get; set; }
    public PatchProperties PatchProperties { get; set; }
```
Patching treats any target property as being a list, so a property can't be set as a simple string value by this method.

The standard custom Properties is not implemented against the Profile so can not be read or set directly.
Adding the property to Profiles
```
    /// <summary>
    /// Custom Properties
    /// </summary>
    public Dictionary<string,object> Properties { get; set; } 
```
appears to fix this and allows properties to be set as string, or as list, directly on the update. e.g.
```
            var update = PatchProfile.Create();
            update.Id = #######
            update.Attributes = new ProfileAttributes() { LastName = "XXXX", FirstName = "David" };

            update.Attributes.Properties = new Dictionary<string, object>() {
                { "$organization", "XXXX Co" },
                { "custom property", "test-prop" },
            }; 
```

**Comment by TIMGATVOLER** [comment]
Any chance we can get an update to the package with these changes?  Custom properties on profiles is something my organization relies on heavily and we recently switched to KlaviyoSharp but I'm struggling to figure out a work around for this.

**Comment by zac-schutt** [comment]
This has been implemented in v1.2.1
